### PR TITLE
[WIP] Enable npm audit and update vulnerable dependencies

### PR DESCRIPTION
**Pull request by AndreyBelym**

### CODE REVIEW

**Comment by AndreyBelym** [comment]
@testcafe-build-bot retest

File: package.json [hunk 1]
```diff
@@ -134,30 +134,32 @@
...
+    "http-server": "^0.11.1",
     "js-yaml": "^3.6.1",
     "license-checker": "^20.0.0",
     "markdownlint": ">=0.0.8",
     "merge-stream": "^1.0.0",
+    "minimist": "^1.2.0",
     "multer": "^1.1.0",
     "opn": "^4.0.2",
-    "publish-please": "^2.1.3",
+    "publish-please": "^3.0.0",
```
Comment (hdorgeval) [inline package.json]: @AndreyBelym : publish-please should be upgraded to version 3.0.1

**Comment by AndreyBelym** [comment]
@testcafe-build-bot retest

File: Gulpfile.js [hunk 2]
```diff
@@ -147,6 +155,43 @@ var PUBLISH_TAG = JSON.parse(fs.readFileSync(path.join(__dirname, '.publishrc'))
 
 var websiteServer = null;
 
+gulp.task('audit', function (done) {
+    let npmCommand = 'npm';
```
Comment (miherlosev) [inline Gulpfile.js]: I propose to start a clean up process for the `gulpfile.js` as mention in https://github.com/DevExpress/testcafe/pull/919.

Create a file `gulp-tasks/npm-audit.js`, move this code in it and run task as following:
```js
const npmAudit = require('gulp-tasks/npm-audit.js');
...

gulp.task('audit', () => {
    // returns Promise
    return npmAudit.perform();
});
```

Comment (miherlosev) [reply Gulpfile.js]: If it makes sense, for running npm commands you can use the [npm-utils](https://www.npmjs.com/package/npm-utils), [npm-audit](https://www.npmjs.com/package/npm-audit-report) or something like that

Comment (kirovboris) [reply Gulpfile.js]: I agree we need extract this code to a separate file, but I am against to refactor `Gulpfile.js` on dozens of small files now.

File: Gulpfile.js [hunk 3]
```diff
@@ -223,11 +250,32 @@ gulp.task('client-scripts-bundle', ['clean'], function () {
...
-gulp.task('server-scripts', ['clean'], function () {
+gulp.step('client-scripts-templates-render', function () {
+    var scripts = [
+        { wrapper: 'src/client/core/index.js.wrapper.mustache', src: 'lib/client/core/index.js' },
+        { wrapper: 'src/client/ui/index.js.wrapper.mustache', src: 'lib/client/ui/index.js' },
+        { wrapper: 'src/client/automation/index.js.wrapper.mustache', src: 'lib/client/automation/index.js' },
+        { wrapper: 'src/client/driver/index.js.wrapper.mustache', src: 'lib/client/driver/index.js' }
+    ];
+
+    return Promise
```
Comment (kirovboris) [inline Gulpfile.js]:
```js
const scriptSrc = fs.readFileSync(script.src).toString();
const filename  = path.basename(script.src);
...
```
etc.

File: Gulpfile.js [hunk 4]
```diff
@@ -438,42 +518,57 @@ gulp.task('clean-website', function () {
...
+gulp.step('put-in-tweets', function () {
     return gulp
         .src('docs/tweets/**/*')
         .pipe(gulp.dest('site/src/_data'));
 });
 
 
-gulp.task('prepare-website', ['put-in-articles', 'put-in-navigation', 'put-in-posts', 'put-in-publications', 'put-in-tweets', 'lint-docs']);
+gulp.step('prepare-website',
+    gulp.parallel(
```
Comment (kirovboris) [inline Gulpfile.js]: It looks terrible, can we refactor this "callback hell" somehow?

File: Gulpfile.js (hunk 2 above)
Comment (miherlosev) [reply Gulpfile.js]: Ok. Let it be

File: Gulpfile.js [hunk 5]
```diff
@@ -550,21 +585,16 @@ function testWebsite (isTravis) {
...
-gulp.task('test-website-travis', function () {
-    return testWebsite(true);
-});
+gulp.task('test-website', gulp.series('build-website-testing', 'serve-website', 'test-website-run'));
+
+gulp.task('test-website-travis', gulp.series('build-website', 'serve-website', 'test-website-run'));
 
-gulp.task('publish-website', ['build-website-production'], function () {
+gulp.step('website-publish-run', function () {
     // NOTE: it's accidentally stopped being compatible with node 0.10 without
```
Comment (miherlosev) [inline Gulpfile.js]: Remove unnecessary comment

**Comment by AndreyBelym** [comment]
@testcafe-build-bot retest

**Comment by AndreyBelym** [comment]
@testcafe-build-bot retest

File: Gulpfile.js [hunk 6]
```diff
@@ -165,7 +185,7 @@ gulp.task('lint', function () {
             'Gulpfile.js'
         ])
         .pipe(eslint())
-        .pipe(eslint.format())
+        .pipe(eslint.format(process.env.ESLINT_FORMATTER))
```
Comment (miherlosev) [inline Gulpfile.js]: Why does it need? We dont't use a custom formatter passed via environment variable.

Comment (AndreyBelym) [reply Gulpfile.js]: I've found that WebStorm can parse links outputted by [unix](https://eslint.org/docs/user-guide/formatters/#unix) or [visualstudio](https://eslint.org/docs/user-guide/formatters/#visualstudio) formatters and navigate to the specific line and column where the error was emitted. 

But the default [stylish](https://eslint.org/docs/user-guide/formatters/#stylish) is more readable for plain text, so I would keep it as the default one, and use the `unix` formatter only in WebStorm.

Comment (miherlosev) [reply Gulpfile.js]: Ok

**Comment by miherlosev** [comment]
ping @kirovboris

**Pull request merged**